Add config props for adaptive remote task req size (#15951)
trinodb/trino: docs/src/main/sphinx/admin/properties-query-management.rst

The amount of time a query is allowed to recover between running out of memory
and being killed, if ``query.low-memory-killer.policy`` or

> ``task.low-memory-killer.policy`` is set to value different than ``none``.

Reviewer: Because this is an unrelated change, this should be pulled out into a separate commit.

query is removed from the query history buffer and no longer available in
the :doc:`/admin/web-interface`.

``query.remote-task.enable-adaptive-request-size``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** :ref:`prop-type-boolean`
* **Default value:** ``true``

> Enables dynamic split of ``taskUpdateRequest`` for schemas featuring large

Reviewer: taskUpdateRequest isn't a user-visible construct. I'd only mention that it's dynamically splitting tasks.

You can also say "large schemas." A schema is the shape of the table, and there's no need to say "schema string."

So my proposal:
"Enables dynamically splitting up server requests sent by tasks, which can prevent out-of-memory errors for large schemas."

strings in table properties.

``query.remote-task.guaranteed-splits-per-task``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** :ref:`prop-type-integer`
* **Default value:** ``3``

> The number of guaranteed splits that will be sent in ``taskUpdateRequest`` for

Reviewer 2: ```suggestion
The number of guaranteed splits that is sent in ``taskUpdateRequest`` for
```

Reviewer: are...

Reviewer 2: Hahah ... one number is.. many splits are .. ;-)

coordinator. If the coordinator is unable to receive updates from a remote task
before this value is reached, the coordinator treats the task as failed.

``query.remote-task.max-request-size``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** :ref:`prop-type-data-size`

> * **Default value:** ``8``

Reviewer 2: thats not a valid data size value

Reviewer: Yeah, this should be 8MB, and you don't need to specify that it's in megabytes. A data size is `2GB` or `16MB` or `10KB`, the user specifies their own units.

> The maximum size of ``taskUpdateRequest`` in megabytes.

Reviewer 2: I dont think this is in megabytes ..

``query.remote-task.request-size-headroom``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** :ref:`prop-type-data-size`

> * **Default value:** ``2``

Reviewer 2: not a valid value

> The headroom available, in megabytes, for ``taskUpdateRequest`` where the

Reviewer 2: not in megabytes

> expected mean value of ``taskUpdateRequest`` will be ``request-size-limit`` -

Reviewer 2: ```suggestion
expected mean value of ``taskUpdateRequest`` is ``request-size-limit`` -
```

``request-size-limit-headroom``.

> 

Reviewer 3: Remove extra line
